src/lib/libaddfunction.js

```diff
@@ -139,27 +139,38 @@ addToLibrary({
   },
 
   $freeTableIndexes: [],
-
+  $usedTableLength: 0,
+  $usedTableLength__postset: () => addAtInit(`usedTableLength = {{{ from64Expr('wasmTable.length') }}}`),
   // Weak map of functions in the table to their indexes, created on first use.
   $functionsInTableMap: undefined,
 
-  $getEmptyTableSlot__deps: ['$freeTableIndexes', '$wasmTable'],
+  $getEmptyTableSlot__deps: ['$freeTableIndexes', '$wasmTable', '$usedTableLength'],
   $getEmptyTableSlot: () => {
     // Reuse a free index if there is one, otherwise grow.
     if (freeTableIndexes.length) {
       return freeTableIndexes.pop();
     }
+    if (usedTableLength < {{{ from64Expr('wasmTable.length') }}}) {
+      return usedTableLength++;
+    }
     // Grow the table
     try {
       /** @suppress {checkTypes} */
-      wasmTable.grow({{{ toIndexType('1') }}});
+      wasmTable.grow({{{ toIndexType('1000') }}});
```

Comment:
I don't love having this hardcoded constant here here, but I also don't want to introduce something like `MEMORY_GROWTH_GEOMETRIC_STEP` that we have for memory. 

Should we perhaps make it a power of 2?  How about 256?

Reply:
I am not sure, if this will be beneficial, I would assume the struct in the UA being already aligned. And I would more lean towards 1024

Reply:
If we do go with a fixed constant like this I think it at least warrants a comment explaining how we came to it.

Reply:
Ok, I have changed to 1024 and added a comment.